src/LaunchDarkly.EventSource/EventSource.cs

```diff
@@ -187,7 +187,11 @@ public async Task StartAsync()
                     if (ReadyState != ReadyState.Shutdown)
                     {
                         Exception realException = e;
-                        if (e is OperationCanceledException oe)
+                        if (realException is AggregateException ae && ae.InnerExceptions.Count == 1)

```

Comment:
A somewhat frustrating feature of the Task API is that if you've done any kind of wrapping or decorating of a task, exceptions that are thrown out of it might appear in their original form, or they might be wrapped in an AggregateException.